src/VectorizeLoops.cpp

```diff
@@ -429,6 +430,19 @@ class VectorizeLoops : public IRMutator {
                 // multiple copies of the if statement.
                 debug(3) << "Scalarizing if then else\n";
                 stmt = scalarize(op);
+
+                // If the condition is likely, we should generate an all-true case.
+                const Call *c = cond.as<Call>();
+                if (c && (c->is_intrinsic(Call::likely) ||
+                          c->is_intrinsic(Call::likely_if_innermost))) {
+                    Expr all_true = const_true();
+                    for (int i = 0; i < lanes; i++) {
+                        all_true = all_true && extract_lane(c->args[0], i);
+                    }
+                    stmt =
+                        IfThenElse::make(Call::make(Bool(), c->name, {all_true}, Call::PureIntrinsic),
```

Comment:
Will scalarize remove the likely intrinsics inside the mutated stmt going into the then case? I would not expect that to be the case and it seems they should be removed.